### PR TITLE
CI: move FreeBSD and Windows tests to GHA

### LINKED ISSUE #2412: [Windows] `TestStart` is failing (`hcs::CreateComputeSystem  ... The requested operation for attach namespace failed`)
```
=== CONT  TestStart
    container_start_test.go:33: assertion failed: 
        Command:  C:\Windows\system32\config\systemprofile\go\bin\nerdctl.exe --namespace=nerdctl-test start nerdctl-teststart
        ExitCode: 1
        Error:    exit status 1
        Stdout:   
        Stderr:   time="2023-08-02T16:37:23Z" level=fatal msg="1 errors:\nfailed to create shim task: hcs::CreateComputeSystem af3db010db23180681e63f7ecfe60eefacf121517b61050416bcdce5c0eabc8e: The requested operation for attach namespace failed.: unknown"
        
        
        Failures:
        ExitCode was 1 expected 0
        Expected stdout to contain "nerdctl-teststart"
        Expected no error
--- FAIL: TestStart (2.54s)
```
https://cirrus-ci.com/task/6205957789712384

Tried 5 times, and still failing

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Fix #2412

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
```
=== CONT  TestStart
    container_start_test.go:33: assertion failed: 
        Command:  C:\Users\runneradmin\go\bin\nerdctl.exe --namespace=nerdctl-test start nerdctl-teststart
        ExitCode: 1
        Error:    exit status 1
        Stdout:   
        Stderr:   time="2023-08-05T12:38:39Z" level=fatal msg="1 errors:\nfailed to create shim task: hcs::CreateComputeSystem 6e491ad2211df7a5b994f7068d24d8b5fe1be99568c99f281d34ae8839dd4921: The requested operation for attach namespace failed.: unknown"
        
        
        Failures:
        ExitCode was 1 expected 0
        Expected stdout to contain "nerdctl-teststart"
        Expected no error
--- FAIL: TestStart (3.72s)
```

https://github.com/containerd/nerdctl/actions/runs/5770671246/job/15643938498?pr=2418

**Comment by AkihiroSuda** [comment]
```
=== RUN   TestRunHostProcessContainer
    container_run_windows_test.go:36: assertion failed: 
        Command:  C:\Users\runneradmin\go\bin\nerdctl.exe --namespace=nerdctl-test run --rm --isolation=host gcr.io/k8s-staging-e2e-test-images/busybox:1.29-2 hostname
        ExitCode: 0
        Stdout:   fv-az269-586
        
        Stderr:   time="2023-08-05T15:18:47Z" level=warning msg="default network named \"nat\" does not have an internal nerdctl ID or nerdctl-managed config file, it was most likely NOT created by nerdctl"
        time="2023-08-05T15:18:48Z" level=warning msg="default network named \"nat\" does not have an internal nerdctl ID or nerdctl-managed config file, it was most likely NOT created by nerdctl"
        {"level":"debug","msg":"[cni-net] Plugin wcn-net version .","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"[net] Network interface: {Index:13 MTU:1500 Name:Ethernet HardwareAddr:60:45:bd:80:58:3c Flags:up|broadcast|multicast|running} with IP addresses: [fe80::b096:8c3:4993:dffb/64 10.1.1.51/16]","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"[net] Network interface: {Index:1 MTU:-1 Name:Loopback Pseudo-Interface 1 HardwareAddr: Flags:up|loopback|multicast|running} with IP addresses: [::1/128 127.0.0.1/8]","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"[net] Network interface: {Index:9 MTU:1500 Name:vEthernet (nat) HardwareAddr:00:15:5d:06:32:ec Flags:up|broadcast|multicast|running} with IP addresses: [fe80::f3dc:de33:1735:99cf/64 172.30.208.1/20]","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"[cni-net] Plugin started.","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"[cni-net] Processing ADD command with args {ContainerID:87338e7a3996ecd864ecdec9d81e6cf789d5be0ebca6272268c824303416fddf Netns:44690802-cf48-4cce-b995-b53386a0ddb1 IfName:eth0 Args:IgnoreUnknown=1 Path:C:\\Program Files\\containerd\\cni\\bin}.","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"[cni-net] Read network configuration \u0026{CniVersion:0.2.0 Name:nat Type:nat Ipam:{Type: Environment: AddrSpace: Subnet:172.30.208.0/12 Address: QueryInterval: Routes:[{Dst:{IP:\u003cnil\u003e Mask:\u003cnil\u003e} GW:172.30.208.1}]} DNS:{Nameservers:[] Domain: Search:[] Options:[]} OptionalFlags:{LocalRoutePortMapping:false AllowAclPortMapping:false ForceBridgeGateway:false} RuntimeConfig:{PortMappings:[] DNS:{Servers:[] Searches:[] Options:[]}} AdditionalArgs:[]}.","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"Parsing port mappings from []","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"[cni-net] Found network 9fc57979-48ef-4b3e-ba92-cd07aa9fb644 with subnet [{{172.30.208.0 fffff000} 172.30.208.1 [{NetworkPolicy [123 34 84 121 112 101 34 58 34 86 76 65 78 34 44 34 83 101 116 116 105 110 103 115 34 58 123 34 73 115 111 108 97 116 105 111 110 73 100 34 58 49 125 125]}]}].","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"[cni-net] Creating a new Endpoint","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"hcn::HostComputeEndpoint::Create id=","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"hcn::HostComputeEndpoint::Create JSON: {\"Name\":\"87338e7a3996ecd864ecdec9d81e6cf789d5be0ebca6272268c824303416fddf_nat\",\"HostComputeNetwork\":\"9fc57979-48ef-4b3e-ba92-cd07aa9fb644\",\"Dns\":{},\"Routes\":[{\"NextHop\":\"172.30.208.1\",\"DestinationPrefix\":\"0.0.0.0/0\"}],\"SchemaVersion\":{\"Major\":2,\"Minor\":0}}","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"hcn::HostComputeEndpoint::AddNamespaceEndpoint id=8e63cf09-edd7-4978-8bfb-8a854e6817f9","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"hcn::HostComputeNamespace::ModifyNamespaceSettings id=4[469](https://github.com/containerd/nerdctl/actions/runs/5771381600/job/15645287234?pr=2418#step:8:470)0802-cf48-4cce-b995-b53386a0ddb1","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"[cni-net] result: IP4:{IP:{IP:172.30.220.167 Mask:fffff000} Gateway:172.30.208.1 Routes:[]}, DNS:{Nameservers:[] Domain: Search:[] Options:[]}","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"[cni-net] Plugin stopped.","time":"2023-08-05T15:18:48Z"}
        time="2023-08-05T15:18:48Z" level=warning msg="networking setup error has occurred" error="<nil>"
        time="2023-08-05T15:18:48Z" level=warning msg="default network named \"nat\" does not have an internal nerdctl ID or nerdctl-managed config file, it was most likely NOT created by nerdctl"
        {"level":"debug","msg":"[cni-net] Plugin wcn-net version .","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"[net] Network interface: {Index:13 MTU:1500 Name:Ethernet HardwareAddr:60:45:bd:80:58:3c Flags:up|broadcast|multicast|running} with IP addresses: [fe80::b096:8c3:4993:dffb/64 10.1.1.51/16]","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"[net] Network interface: {Index:1 MTU:-1 Name:Loopback Pseudo-Interface 1 HardwareAddr: Flags:up|loopback|multicast|running} with IP addresses: [::1/128 127.0.0.1/8]","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"[net] Network interface: {Index:9 MTU:1500 Name:vEthernet (nat) HardwareAddr:00:15:5d:06:32:ec Flags:up|broadcast|multicast|running} with IP addresses: [fe80::f3dc:de33:1735:99cf/64 172.30.208.1/20]","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"[cni-net] Plugin started.","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"[cni-net] Processing DEL command with args {ContainerID:87338e7a3996ecd864ecdec9d81e6cf789d5be0ebca6272268c824303416fddf Netns:44690802-cf48-4cce-b995-b53386a0ddb1 IfName:eth0 Args:IgnoreUnknown=1 Path:C:\\Program Files\\containerd\\cni\\bin}","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"[cni-net] Read network configuration \u0026{CniVersion:0.2.0 Name:nat Type:nat Ipam:{Type: Environment: AddrSpace: Subnet:172.30.208.0/12 Address: QueryInterval: Routes:[{Dst:{IP:\u003cnil\u003e Mask:\u003cnil\u003e} GW:172.30.208.1}]} DNS:{Nameservers:[] Domain: Search:[] Options:[]} OptionalFlags:{LocalRoutePortMapping:false AllowAclPortMapping:false ForceBridgeGateway:false} RuntimeConfig:{PortMappings:[] DNS:{Servers:[] Searches:[] Options:[]}} AdditionalArgs:[]}.","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"Parsing port mappings from []","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"hcn::HostComputeNamespace::RemoveNamespaceEndpoint id=8e63cf09-edd7-4978-8bfb-8a854e6817f9","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"hcn::HostComputeNamespace::ModifyNamespaceSettings id=44690802-cf48-4cce-b995-b53386a0ddb1","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"hcn::HostComputeEndpoint::Delete id=8e63cf09-edd7-4978-8bfb-8a854e6817f9","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"[cni-net] DEL succeeded.","time":"2023-08-05T15:18:48Z"}
        {"level":"debug","msg":"[cni-net] Plugin stopped.","time":"2023-08-05T15:18:48Z"}
        
        
        Failures:
        Expected stdout to contain "fv-az269-586\n"
--- FAIL: TestRunHostProcessContainer (0.84s)
```


https://github.com/containerd/nerdctl/actions/runs/5771381600/job/15645287234?pr=2418

**Comment by AkihiroSuda** [comment]
cc @gabriel-samfira @aznashwan FYI

The CI is now green for Windows

File: Vagrantfile.freebsd [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+# Vagrantfile for FreeBSD
+Vagrant.configure("2") do |config|
+  config.vm.box = "generic/freebsd13"
+
+  memory = 2048
+  cpus = 1
```
Comment (fuweid) [inline Vagrantfile.freebsd:23]: do we need two cpus here?

**Review by fuweid — APPROVED** [review]
LGTM

File: Vagrantfile.freebsd (hunk 1 above)
Comment (AkihiroSuda) [reply Vagrantfile.freebsd:23]: Why?

Comment (fuweid) [reply Vagrantfile.freebsd:23]: Because I see the original one (.cirrus.yml) was using two cpus 😂

**Pull request merged**